build/lib/builtInExtensions.ts

```diff
@@ -46,18 +46,27 @@ const webBuiltInExtensions = <IExtensionDefinition[]>productjson.webBuiltInExten
 const controlFilePath = path.join(os.homedir(), '.vscode-oss-dev', 'extensions', 'control.json');
 const ENABLE_LOGGING = !process.env['VSCODE_BUILD_BUILTIN_EXTENSIONS_SILENCE_PLEASE'];
 
```

Comment:
@jmcphers I tried to keep my changes concise in this file in attempt to make pulling in upstream as painless as possible, but I'm not sure I did a good job of that. Do you think I should further refactor it to separate out the Positron code?

Reply:
Based on an informal look I do think more refactoring would be beneficial. Specifically it looks like a lot of functions now take a 'bootstrap' flag so a lot of signatures have changed and there's new code in each function body to handle the bootstrap case.

What I'd suggest is having a different set of functions entirely for bootstrapped extensions -- it will mean duplicating some code but I think it's worth the tradeoff since otherwise our logic gets pretty entangled and difficult to reconcile with upstream changes. 

LMK if you feel differently after giving it a try!

Reply:
Ok, I've refactored this to remove most of the booleans, but I kept one that I think is straightforward enough. Let me know what you think!